modules/reindex/src/internalClusterTest/java/org/opensearch/index/codec/MultiCodecReindexIT.java

```diff
@@ -34,14 +35,10 @@
 import java.util.stream.IntStream;
 
 import static java.util.stream.Collectors.toList;
-import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_BLOCKS_METADATA;
-import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_BLOCKS_READ;
-import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_BLOCKS_WRITE;
-import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_READ_ONLY;
-import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_READ_ONLY_ALLOW_DELETE;
 import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
 import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertNoFailures;
 
+@ClusterScope(scope = Scope.SUITE, numDataNodes = 1, numClientNodes = 0, supportsDedicatedMasters = false)
```

Comment:
@r1walz looks like the flaky-ness is related to connection issue. 
Are we able to consistently solve the issue with ```ClusterScope(scope = Scope.SUITE, numDataNodes = 1, numClientNodes = 0, supportsDedicatedMasters = false)```?

Reply:
My [other test](https://github.com/opensearch-project/OpenSearch/blob/73bbeb57a8a56de0d1f9cf69e711a8a0ff26afe5/server/src/internalClusterTest/java/org/opensearch/nodestats/NodeStatsIT.java#L41) is holding good. So, it should work fine here as well. There's no node-to-node communication so no chance of flakiness coming from there.

Reply:
Thanks a lot for looking into this issue @r1walz , but I honestly doubt that we've identified the fix.  If you look on test hierarchy, there are dozens or so test cases that inherit from `ReindexTestCase`, I would expect the issue to be widespread but it is not. 

The first thing that looks wrong to me is that `MultiCodecReindexIT` is in a wrong place and test task: it should be renamed to `MultiCodecReindexTests` and moved from `internalClusterTest` to `src/test` folder, where all other tests are placed. 

The connectivity could be an issue sometimes and even with 1 node it could happen.

Reply:
@reta @peternied I've a question, is there a way to check the flaky-ness of a test across multiple builds?
I want to understand in what builds it has been been failing in the past month or so, in order to deep dive into the issue. Apologies if this information is shared someplace and I missed it. Thankyou

Reply:
Good question, I am not aware of a system to track test failures over time. 

@andrross do you know of one?

Reply:
I do have [this script](https://gist.github.com/andrross/ee07a8a05beb63f1173bcb98523918b9) which, while far from ideal, is an example of how to crawl the Jenkins output to find failing tests over time. Feel free to adapt to your needs.

Reply:
Here are `gradle-check` runs which failed for `MultiCodecReindexIT.*` (between 24398 and 30125). Thanks for the script, @andrross.

```sh
21 org.opensearch.index.codec.MultiCodecMergeIT.testForceMergeMultipleCodecs (24465,24696,24733,24738,24749,24750,24765,24781,24792,24810,24848,24853,24860,24861,24862,24863,24864,24875,24876,24892,25084)
15 org.opensearch.index.codec.MultiCodecReindexIT.classMethod (24434,24735,24762,24810,24845,24863,24881,24902,24935,24985,25002,25013,25076,25097,27968)
15 org.opensearch.index.codec.MultiCodecReindexIT.testReindexingMultipleCodecs (24434,24735,24762,24810,24845,24863,24881,24902,24935,24985,25002,25013,25076,25097,28473)
```

cc: @sarthakaggarwal97

Reply:
@ankitkala that's correct, ```index.merge.policy.max_merged_segment``` was configured to ```1b``` to disable merges. The surprising part is even after these settings, the segments are getting merged. Do we know why would that be the case?

Reply:
@r1walz thanks for pulling out the builds where the test failed. It looks like the [last failed build](https://build.ci.opensearch.org/job/gradle-check/28473/) due to this test is around a month back. (10/19)

Reply:
> @ankitkala that's correct, `index.merge.policy.max_merged_segment` was configured to `1b` to disable merges. The surprising part is even after these settings, the segments are getting merged. Do we know why would that be the case?

Not sure. I think it's not getting honored somehow. Will need to check more on this. 

Here are the segments after each doc insertion. 

```
[2023-11-17T11:21:00,358][INFO ][o.o.i.c.MultiCodecReindexIT] [testReindexingMultipleCodecs] Segments after ingest [Segment{name='_0', generation=0, committed=true, search=true, sizeInBytes=3402, docCount=5, delDocCount=0, version='9.8.0', compound=true, mergeId='null', attributes={Lucene90StoredFieldsFormat.mode=BEST_SPEED}}]
[2023-11-17T11:21:01,306][INFO ][o.o.i.c.MultiCodecReindexIT] [testReindexingMultipleCodecs] Segments after ingest [Segment{name='_0', generation=0, committed=true, search=true, sizeInBytes=3402, docCount=5, delDocCount=0, version='9.8.0', compound=true, mergeId='null', attributes={Lucene90StoredFieldsFormat.mode=BEST_SPEED}}, Segment{name='_1', generation=1, committed=true, search=true, sizeInBytes=3416, docCount=5, delDocCount=0, version='9.8.0', compound=true, mergeId='null', attributes={Lucene90StoredFieldsFormat.mode=BEST_COMPRESSION}}]
[2023-11-17T11:21:02,253][INFO ][o.o.i.c.MultiCodecReindexIT] [testReindexingMultipleCodecs] Segments after ingest [Segment{name='_0', generation=0, committed=true, search=true, sizeInBytes=3402, docCount=5, delDocCount=0, version='9.8.0', compound=true, mergeId='null', attributes={Lucene90StoredFieldsFormat.mode=BEST_SPEED}}, Segment{name='_1', generation=1, committed=true, search=true, sizeInBytes=3416, docCount=5, delDocCount=0, version='9.8.0', compound=true, mergeId='null', attributes={Lucene90StoredFieldsFormat.mode=BEST_COMPRESSION}}, Segment{name='_2', generation=2, committed=true, search=true, sizeInBytes=3424, docCount=5, delDocCount=0, version='9.8.0', compound=true, mergeId='null', attributes={Lucene90StoredFieldsFormat.mode=BEST_COMPRESSION}}]
[2023-11-17T11:21:03,274][INFO ][o.o.i.c.MultiCodecReindexIT] [testReindexingMultipleCodecs] Segments after ingest [Segment{name='_4', generation=4, committed=true, search=true, sizeInBytes=3362, docCount=4, delDocCount=0, version='9.8.0', compound=true, mergeId='null', attributes={Lucene90StoredFieldsFormat.mode=BEST_SPEED}}, Segment{name='_5', generation=5, committed=true, search=true, sizeInBytes=4047, docCount=16, delDocCount=0, version='9.8.0', compound=false, mergeId='null', attributes={Lucene90StoredFieldsFormat.mode=BEST_SPEED}}]
```

Reply:
I've used the `index.merge.enabled` property instead of setting `index.merge.policy.max_merged_segment`. Also, I've moved the test to `test` folder from `internalClusterTest` folder and renamed it to `MultiCodecReindexTests` from `MultiCodecReindexIT`.

Reply:
@ankitkala here's the thread. I had resolved it earlier.